stream_alert/classifier/payload/payload_base.py

```diff
@@ -62,14 +62,24 @@ def __len__(self):
         )
 
     def __repr__(self):
+        record_data = self._record_data
+        invalid_records = self.invalid_records
+
         try:
-            record_data = json.dumps(self._record_data)
-            invalid_records = json.dumps(self.invalid_records)
+            if isinstance(record_data, dict):
+                record_data = json.dumps(record_data)
         except (TypeError, ValueError):
-            record_data = self._record_data
-            invalid_records = self.invalid_records
             LOGGER.debug('A PayloadRecord has data that is not serializable as JSON')
 
+        try:
+            #Build a JSON string if data is a dict or non-empty list of dicts
+            if (isinstance(invalid_records, dict) or
+                    (isinstance(invalid_records, list) and
+                     invalid_records and isinstance(invalid_records[0], dict))):
```

Comment:
I would not do the `invalid_records[0]` check here and simply do `isinstance(invalid_records, list)`. Nested type checking is a really tough problem which isn't really worth solving.

Reply:
I agree with you on this. However in testing I found some situations/test cases where the underlying data is a list of strings. Dumping this as a json string caused some weird encoding.

Reply:
Can you show me the exact case that's causing the issue? It sounds like a good candidate for writing a unit test

Reply:
@Ryxias This test: test_repr_invalid_records
And you end up with [{\'key\': \'value\'}] instead of [{"key": "value"}] if you don't check one level down.

Reply:
@Ryxias In this case it is a list of dicts not a list of strings so it should be {"key": "value"} not {\'key\': \'value\'}. So, I DO want to encode as JSON if the object inside the list is a dict but not if its a string.